src/main/resources/data-product-definition-schema.json

```diff
@@ -0,0 +1,262 @@
+        { "$ref": "#/definitions/policy-condition-nomatch" },
+        { "$ref": "#/definitions/policy-condition-like" },
+        { "$ref": "#/definitions/policy-condition-between" }
+      ]
+    },
+
+    "policy-condition-match": {
+      "type": "object",
+      "properties": {
+        "match": {
+          "type": "array",
+          "items": { "type": "string" }
+        }
+      }
+    },
+
+    "policy-condition-nomatch": {
+      "type": "object",
+      "properties": {
+        "notequals": {
+          "type": "array",
+          "items": { "type": "string" }
+        }
+      }
+    },
+
+    "policy-condition-like": {
+      "type": "object",
+      "properties": {
+        "like": {
+          "type": "array",
+          "items": { "type": "string" }
+        }
+      }
+    },
+
+    "policy-condition-between": {
+      "type": "object",
+      "properties": {
+        "between": {
+          "type": "object",
+          "required": ["value", "start", "end"],
+          "properties": {
+            "value": { "type": "string" },
+            "start": { "type": "string" },
+            "end": { "type": "string" }
+          }
+        }
+      }
+    },
+
+    "report": {
+      "type": "object",
+      "required": ["id", "name", "version", "dataset", "render", "specification"],
+      "properties": {
+        "id": { "type": "string" },
+        "name": { "type": "string" },
+        "description": { "type": "string" },
+        "created": { "type": "string" },
+        "modified": { "type": "string" },
+        "withdrawn": { "type": "string" },
+        "version": { "type": "string" },
+        "dataset": { "type": "string" },
+        "render": { "type": "string" },
+        "classification": { "type": "string" },
+        "policy": { "type": "array", "items": { "$ref": "#/definitions/policy" } },
+        "specification": { "$ref": "#/definitions/report-specification" }
+      }
+    },
+
+    "report-specification": {
+      "type": "object",
+      "required": ["template", "field"],
+      "properties": {
+        "template": { "enum": ["list"] },
+        "field": {
+          "type": "array",
+          "items": { "$ref": "#/definitions/report-field" }
+        }
+      }
+    },
+
+    "report-field": {
+      "type": "object",
+      "required": ["name"],
+      "properties": {
+        "description": { "type": "string" },
+        "name": { "type": "string" },
+        "display": { "type": "string" },
+        "formula": { "type": "string" },
+        "visible": { "enum": ["true", "false", "mandatory"] },
+        "sortable": { "type": "boolean" },
+        "defaultsort": { "type": "boolean" },
+        "wordwrap": { "enum": ["none", "normal", "break-words"] },
+        "alias": { "$ref": "#/definitions/alias" },
+        "filter": { "$ref": "#/definitions/filter" },
+        "type": { "enum": ["boolean", "int", "long", "float", "double", "string", "date", "time", "datetime", "timestamp"] }
```

Comment:
Is this schema meant to be the same as https://github.com/ministryofjustice/hmpps-digital-prison-reporting-lib/blob/main/src/main/resources/data-product-definition-schema.json?
There are some divergences between them. For example the "type" in the library schema also has "null', "bytes" and "timestamp". 
Also "wordwrap" in the library has "on", "off", "none".
And "visible" in the library is outdated. 
I suggest after we bring the format up to date to keep only one schema file in one repository to avoid one of them falling behind.
It might be better to keep it in the library as clients would also be interested in this schema file. 
And in the definitions repository to put a link in the Readme. 
But since it is used in the test for the definitions validation this might be a problem to remove it.

Reply:
In terms of those changes to the schema, yes that's exactly right:
<img width="933" alt="Screenshot 2024-03-22 at 08 35 04" src="https://github.com/ministryofjustice/hmpps-dpr-data-product-definitions/assets/110236995/9d32c493-fd1f-487d-ac42-6d6beef4b717">

I'm just waiting on Dominic to confirm before I merge these changes.

Reply:
In terms of the library being outdated; I think the other PR should solve all of these.